scipy/sparse/_coo.py

```diff
@@ -394,8 +402,14 @@ def tocsr(self, copy=False):
             M,N = self.shape
             idx_dtype = get_index_dtype((self.row, self.col),
                                         maxval=max(self.nnz, N))
-            row = self.row.astype(idx_dtype, copy=False)
-            col = self.col.astype(idx_dtype, copy=False)
+            xp, array_api = get_namespace(self.row, self.col)
+            if array_api:
+                # TODO: Removed dtype, unsupported
+                row = xp.astype(self.row, idx_dtype)
+                col = xp.astype(self.col, idx_dtype)
+            else:
+                row = self.row.astype(idx_dtype, copy=False)
+                col = self.col.astype(idx_dtype, copy=False)
```

Comment:
The resulting data structure will have `array_api` `row` and `col`, but also have `indptr` and `indices` as `NumPy` arrays.